examples/ws_handlers.hpp

```diff
@@ -0,0 +1,128 @@
+
+#pragma once
+
+#include <spdlog/fmt/fmt.h>
+
+#include <malloy/utils.hpp>
+#include <malloy/http/request.hpp>
+#include <malloy/error.hpp>
+
+#include <malloy/websocket/connection.hpp>
+
+#include <thread>
+
+namespace malloy::examples::ws {
+template<bool isClient>
+using ws_connection = std::shared_ptr<malloy::websocket::connection<isClient>>;
+
+
+template<bool isClient>
+void oneshot_read(const ws_connection<isClient>& conn, std::invocable<malloy::error_code, std::string> auto&& on_read) {
+    auto buffer = std::make_shared<boost::beast::flat_buffer>();
+    conn->read(*buffer, [buffer, on_read = std::forward<decltype(on_read)>(on_read)](auto ec, auto) {
+        on_read(ec, malloy::buffers_to_string(malloy::buffer(buffer->cdata(), buffer->size())));
+    });
+}
+
+template<bool isClient>
+void oneshot(const auto& req, ws_connection<isClient> conn, const std::string& msg_data) {
+    auto msg = std::make_shared<std::string>(msg_data); // Keep message memory alive
+    conn->accept(req, [msg, conn] { conn->send(malloy::buffer(*msg), [msg](auto, auto) {}); });
+}
+
+template<bool isClient>
```

Comment:
This could use some short comment describing it's functionality.

Reply:
I think it should really be named `oneshot_send` actually, I'll rename it